river/main.zig

```diff
@@ -140,6 +140,11 @@ fn defaultInitPath() !?[:0]const u8 {
     };
 
     os.accessZ(path, os.X_OK) catch |err| {
+        if (err == error.PermissionDenied) blk: {
+            os.accessZ(path, os.R_OK) catch break :blk;
+            std.log.err("failed to run init executable {s}: the file is not executable", .{path});
+            os.exit(1);
+        }
```

Comment:
Sorry for the bikeshedding, but this `break :blk` control flow feels a bit convoluted. I think an `if` statement would be better here:
```suggestion
        if (err == error.PermissionDenied) {
            if (os.accessZ(path, os.R_OK)) {
                std.log.err("failed to run init executable {s}: the file is not executable", .{path});
                os.exit(1);
            } else |_| {}
        }
```
Other than that this looks ready to merge :)

Reply:
Huh, I always forget `if` works on error unions as well.